lwt_retry.opam

```diff
@@ -0,0 +1,31 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Utilities for retrying Lwt computations"
+maintainer: [
+  "Raphaël Proust <code@bnwr.net>" "Anton Bachin <antonbachin@yahoo.com>"
+]
+authors: ["Jérôme Vouillon" "Jérémie Dimino"]
```

Comment:
I think you can give credit where credit is due (yourself? some colleagues of yours?)

Reply:
Whoops! Oversight here. Sorry! Thanks for catching this :)

Reply:
Ah, this is just generated based on the dune-project, which is why I missed.

Reply:
Hmm, this is actually trickier than I'd expect, as dune doesn't support `maintainers` or `authors` fields on `package` stanzas until dune 2.0. But there are several other aspects of the lwt dune config that are incompatible with dune 2.0. Most are trivial so far, but some have to do with the C FFI:

```
File "test/retry/dune", lines 8-9, characters 1-33:
8 |  (action
9 |   (run %{exe:main.exe})))
File "_build/.dune/default/src/unix/dune", line 178, characters 1-43:
178 |  (c_flags -I. (:include unix_c_flags.sexp))
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: 'c_flags' was deleted in version 2.0 of the dune language. Use the
(foreign_stubs ...) field instead.
Had 1 error, waiting for filesystem changes...
```

So I think our options here are:

1. Leave the maintainer/authors package as is, generated from the data in the dune-project file.
2. use an opam.template file for the retry package.
3. Update lwt to use dune 2.0, and put the data in the `package` stanza.

3 seems like the right (tm) way to do this to me. If you agree, I can make that change in a separate PR then rebase this on top. WDYT?

Reply:
For 3 I need to first check the dependency cone for dune.2+ (minimal ocaml version and other such minimal version constraints). One of the thing is we try to keep maximum compatiblity because we treat Lwt as a kind of foundational library.

Depending on the compatibility, I'd go for either 2 or 3. I can check that some times in the coming week.

Reply:
Dune doesn't have external dependencies. It requires (at least) ocaml 4.08 and has a fallback through a secondary ocaml compiler for earlier versions. I don't think there's anything to be concerned here with.

Reply:
Thank, @MisterDA . Opened a PR for the update at https://github.com/ocsigen/lwt/pull/1035